bonsai-core/src/main/scala/com/stripe/bonsai/layout/Product3Layout.scala

```diff
@@ -0,0 +1,95 @@
+package com.stripe.bonsai
+package layout
+
+import java.io.{ DataOutput, DataInput }
+
+class Product3Layout[A, B, C, D](
```

Comment:
In the long run do you think we'll need to codegen these out for larger products?